### ISSUE TITLE
Crop the WSI

### ISSUE THREAD

**Issue by enochsol** [BMIRDS/deepslide#41]
Before the Pathologists annotate the crops, how did you crop the whole slide image? Is there any tool used?

**Comment by enochsol** [comment]
Thank you so much! That would help.

For example, if I have just 3 slides and 2 labels(tumor and benign):
slide 1 has 10 annotations (5 tumor and 5 benign)
slide 2 has 8 annotations (5 tumor and 3 benign)
slide 3 has 6 annotations (5 tumor and 1 benign)

Then finally I will have 10+8+6 (24 (do we call it as a patches? )) training dataset with 15 tumor and 9 benign classes, is that the right assumption? 

If so the first steps is converting the whole slide images (since it is very high resolutions) into patches with the above procedure?
After this step we can start using the Deepslide library by following the readme, right?

**Comment by JosephDiPalma** [comment]
You wouldn't have 24 patches, but rather 24 annotations.

Once you have the annotated regions, you can convert them into patches using the file `2_process_patches.py`.

**Comment by enochsol** [comment]
Now it is getting more clear.
Thus, the steps would be:
1. Convert the slides to annotated images (it will removes uninterested areas from the slide and keep save the annotations as own images)
2. Then convert the annotated images to patches (like 256 by 256 pixel) since the annotated images still might be in a high resolutions (like thousands by thousands). For this step we may use 2_process_patches.py

Please could confirm

**Comment by JosephDiPalma** [comment]
That is correct.

**Comment by enochsol** [comment]
Thank you so much!! Crystal clear :)

**Comment by enochsol** [comment]
What are the steps for model validation? 

And also how is the visualization? Patch level or Slide level visualization?

**Comment by JosephDiPalma** [comment]
> What are the steps for model validation?

If by model validation, you mean producing outputs for an unseen dataset (i.e., a test set), we recommend following the instructions on the [Section 4](https://github.com/BMIRDS/deepslide#4-testing-on-wsi) of the README.

> And also how is the visualization? Patch level or Slide level visualization?

We create the visualization at the patch-level because the model operates on patches. However, our visualization code combines the outputs over all patches and plots them at the slide-level.

**Comment by enochsol** [comment]
> > What are the steps for model validation?
> 
Thank you!
> If by model validation, you mean producing outputs for an unseen dataset (i.e., a test set), we recommend following the instructions on [Section 4](https://github.com/BMIRDS/deepslide#4-testing-on-wsi) of the README.
> 
> > And also how is the visualization? Patch level or Slide level visualization?
> 
> We create the visualization at the patch-level because the model operates on patches. However, our visualization code combines the outputs over all patches and plots them at the slide-level.
How it is combining and draw the bounding box around?

**Comment by JosephDiPalma** [comment]
> How it is combining and draw the bounding box around?

We don't actually plot the bounding boxes on each patch individually and then combine them. As we are using a sliding window approach, we take the entire slide and use indexing to pass in the patches to the network. Then we draw the bounding boxes on the slide corresponding to the predicted class. See [Line 286](https://github.com/BMIRDS/deepslide/blob/74616480decb714e988e642871a42a018b744695/code/utils_evaluation.py#L286) in `code/utils_evaluation.py` for further information.

**Comment by enochsol** [comment]
Thank you so much!

What meant by "a sliding window approach"? Please could you compare it with the below approach? 

Based on our pervious discussion:
the steps to get patches would be:

1. Crop the annotated images part from the whole slide (it will removes uninterested areas from the slide and save the annotated parts as their own images)
2. Then convert the annotated images to patches (like 256 by 256 pixel) since the annotated images still might be in a high resolutions (like thousands by thousands). For this step we may use 2_process_patches.py

**Comment by enochsol** [comment]
Thank you so much!!
